android/app/src/main/java/app/organicmaps/car/NavigationSession.java

```diff
@@ -191,28 +210,46 @@ public void onCompassUpdated(double north)
   public void onDisplayChanged(@NonNull final DisplayType newDisplayType)
   {
     Logger.d(TAG);
-    final boolean isUsedOnDeviceScreenShown = mScreenManager.getTop() instanceof MapPlaceholderScreen;
+    final Screen screen = mScreenManager.getTop();
+    final boolean isMapPlaceholderScreenShown = screen instanceof MapPlaceholderScreen;
+    final boolean isPermissionsOrErrorScreenShown = screen instanceof RequestPermissionsScreen || screen instanceof ErrorScreen;
     final RoutingController routingController = RoutingController.get();
     if (newDisplayType == DisplayType.Car)
     {
-      LocationState.nativeSetListener(this);
-      onMyPositionModeChanged(LocationState.nativeGetMode());
-      mScreenManager.popToRoot();
+      onStart(this);
+      mSurfaceRenderer.enable();
+
+      // If we have Permissions or Error Screen in Screen Manager (either on the top of the stack or after MapPlaceholderScreen) do nothing
+      if (isPermissionsOrErrorScreenShown)
```

Comment:
a special hack just for permission screen?

Reply:
Yes, it's important not to disrupt the logic.

1. The Permissions Request screen is displayed.
2. Switch to the phone screen.
3. Go back to AA.
4. At this point, we should display the Permissions Request screen, not the Map Screen.

Reply:
My concern that it introduces this new special state "isPermissionsOrErrorScreenShown" which should be handled in multiple places...